jersey/src/main/java/resource/CounterResource.java

```diff
@@ -0,0 +1,59 @@
+package resource;
+
+import dao.CounterDAO;
+import dao.ICounterDAO;
+import jakarta.inject.Singleton;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+@Singleton
+@Path("/counter")
+public class CounterResource {
+
+    private final ICounterDAO counterDAO = CounterDAO.getInstance();
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response get() {
+        int value = counterDAO.getCounter();
+        String date = DateTimeFormatter.ISO_INSTANT.format(Instant.now());
+        CounterDTO counterDTO = new CounterDTO(value, date);
+        return Response.ok().entity(counterDTO).build();
+    }
+
+    @POST
+    public Response post() {
+        counterDAO.incrementCounter();
+        return Response.ok().build();
+    }
+
+    @DELETE
+    public Response delete(@HeaderParam("Subtraction-Value") String subtractionValue) {
+        try {
+            counterDAO.subtractCounter(Integer.parseInt(subtractionValue));
+        } catch (NumberFormatException e) {
+            Response.status(HttpServletResponse.SC_PRECONDITION_FAILED);
```

Comment:
Интересно, даже забыл про такой заголовок :)
Но кажется лучше тут все таки просто 400 вернуть

`
This happens with conditional requests on methods other than GET or HEAD when the condition defined by the If-Unmodified-Since or If-None-Match headers is not fulfilled
`